src/test/lib/optimizer/optimizer_test.cpp

```diff
@@ -84,14 +104,39 @@ TEST_F(OptimizerTest, AssertsInPlanReferences) {
   // clang-format off
```

Comment:
The test checks that we throw an error if an expression used by a node (`x` in that case) comes from the same LQP. Tested code is here:
https://github.com/hyrise/hyrise/blob/5adcbd2bbae08c1150eb277fd99bfe3cc788370e/src/lib/optimizer/optimizer.cpp#L94-L110

Since the checks drill down if the original node referenced by the LQPColumnExpression is in the same LQP, it's kind of what it says. Maybe `AssertsOriginalNodeInSamePlan`?

Now to the second question: It worked before. However, the used uncorrelated subquery contains `node_b`, which outputs `x`. Since we treat uncorrelated subqueries as regular nodes, they are now also part of the mapping LQP root -> all nodes in the LQP. Thus, it would not throw an error anymore.

Reply:
`AssertsOriginalNodeInSamePlan` sounds better to me.